### PR TITLE
Add documentation for Component SLIs feature

### PR DESCRIPTION
**Pull request by logicalhan**
This PR provides documentation for the [Component SLIs feature](https://github.com/kubernetes/enhancements/issues/3466).

### CODE REVIEW

**Comment by jihoon-seo** [comment]
/remove-area blog release-eng
/remove-language de es fr hi id it ja ko pl pt ru vi zh
/remove-sig release

File: content/en/docs/reference/instrumentation/slis.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+weight: 20
+---
+
+<!-- overview -->
+
+{{< feature-state for_k8s_version="v1.26" state="alpha" >}}
+
+The Component SLIs feature allows enabling of a new metrics endpoint for each Kubernetes
+binary. This metric endpoint is exposed on the standard port of each binary at the endpoint
+`metrics/slis`.
```
Comment (sftim) [inline content/en/docs/reference/instrumentation/slis.md:15]: Is this right?
```suggestion
`/metrics/slis`. 
```

If not, readers may assume that the path is `/metrics/slis`

File: content/en/docs/reference/instrumentation/slis.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+The Component SLIs feature allows enabling of a new metrics endpoint for each Kubernetes
+binary. This metric endpoint is exposed on the standard port of each binary at the endpoint
+`metrics/slis`. 
+
+<!-- body -->
+
+## SLI Metrics
+
+We expose two metrics; a gauge (which represents the current state of the healthcheck) and a
+counter (which records the cumulative counts that we've observed for each healthcheck state).
```
Comment (sftim) [inline content/en/docs/reference/instrumentation/slis.md:22]: (nit)

We [avoid using we](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-using-we)

Is this right?
```suggestion
With SLI metrics enabled, each Kubernetes component exposes two metrics,
labelled per healthcheck:

- a gauge (which represents the current state of the healthcheck)
- a counter (which records the cumulative counts observed for each healthcheck state)
```

File: content/en/docs/reference/instrumentation/slis.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+   healthchecksTotal = k8smetrics.NewCounterVec(
+      &k8smetrics.CounterOpts{
+         Namespace:      "kubernetes",
+         Name:           "healthchecks_total",
+         Help:           "This metric records the results of all healthcheck.",
+         StabilityLevel: k8smetrics.ALPHA,
+      },
+      []string{"name", "type", "status"},
+   )
+)
```
Comment (sftim) [inline content/en/docs/reference/instrumentation/slis.md:47]: It's OK to omit the source code; we don't expect readers to know how to code in Golang. We could hyperlink to the implementation if that's important.

Please either omit this or make it clear why we should include it.

File: content/en/docs/reference/instrumentation/slis.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+```
+
+The gauge data looks like this:
+
+```shell
```
Comment (sftim) [inline content/en/docs/reference/instrumentation/slis.md:52]: This is actually OpenMetrics (or something close), not shell. It's OK to omit a highlighting hint. Please don't set one that's wrong though.

File: content/en/docs/reference/instrumentation/slis.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+kubernetes_healthcheck{name="informer-sync",type="readyz"} 1
+kubernetes_healthcheck{name="log",type="healthz"} 1
+kubernetes_healthcheck{name="log",type="readyz"} 1
+kubernetes_healthcheck{name="ping",type="healthz"} 1
+kubernetes_healthcheck{name="ping",type="readyz"} 1
+```
+
+While the counter data looks like this:
+
+```shell
```
Comment (sftim) [inline content/en/docs/reference/instrumentation/slis.md:69]: (nit)
This is actually OpenMetrics (or something close), not shell. It's OK to omit a highlighting hint. Please don't set one that's wrong though.

File: content/en/docs/reference/instrumentation/slis.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+kubernetes_healthchecks_total{name="log",status="success",type="readyz"} 15
+kubernetes_healthchecks_total{name="ping",status="success",type="healthz"} 15
+kubernetes_healthchecks_total{name="ping",status="success",type="readyz"} 15
+```
+
+## Using this data
+
+The component SLIs metrics endpoint is intended to be scraped at a high frequency. Scraping
+at a high frequency means that you end up with greater granularity of the gauge's signal, which
+can be then used to calculate SLOs. The `metrics/slis` endpoint provides the raw data necessary
```
Comment (sftim) [inline content/en/docs/reference/instrumentation/slis.md:90]:
```suggestion
can be then used to calculate SLOs. The `/metrics/slis` endpoint provides the raw data necessary
```

File: content/en/docs/reference/instrumentation/slis.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+to calculate an availability SLO for the respective Kubernetes component.
```
Comment (sftim) [inline content/en/docs/reference/instrumentation/slis.md:91]: (nit)
What's a high frequency? Are you thinking tens of scrapes per minute, or tens of scrapes per second?

File: content/en/docs/reference/instrumentation/slis.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
+---
+reviewers:
+- logicalhan
+title: Kubernetes Component SLIs
```
Comment (sftim) [inline content/en/docs/reference/instrumentation/slis.md:4]: (nit)
```suggestion
title: Kubernetes Component SLI Metrics
linkTitle: Service Level Indicator Metrics
```

File: content/en/docs/reference/instrumentation/slis.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+content_type: reference
+weight: 20
+---
+
+<!-- overview -->
+
+{{< feature-state for_k8s_version="v1.26" state="alpha" >}}
+
+The Component SLIs feature allows enabling of a new metrics endpoint for each Kubernetes
```
Comment (sftim) [inline content/en/docs/reference/instrumentation/slis.md:13]: (nit)
```suggestion
As an alpha feature, Kubernetes lets you configure Service Level Indicator (SLI) metrics for each
Kubernetes component
```

File: content/en/docs/reference/instrumentation/slis.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+binary. This metric endpoint is exposed on the standard port of each binary at the endpoint
```
Comment (sftim) [inline content/en/docs/reference/instrumentation/slis.md:14]: (nit)
```suggestion
binary. This metric endpoint is exposed on the serving HTTPS port of each component,
at the path
```

File: content/en/docs/reference/instrumentation/slis.md [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+`metrics/slis`. 
+
+<!-- body -->
+
+## SLI Metrics
+
+We expose two metrics; a gauge (which represents the current state of the healthcheck) and a
+counter (which records the cumulative counts that we've observed for each healthcheck state).
+
```
Comment (sftim) [inline content/en/docs/reference/instrumentation/slis.md:23]: (nit)
```suggestion

You can use the metric information to calculate per-component availability statistics.
For example, the API server checks the health of etcd. You can work out and report how
available or unavailable etcd has been - as reported by its client, the API server.

```

File: content/en/docs/reference/instrumentation/slis.md [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+---
+
+<!-- overview -->
+
+{{< feature-state for_k8s_version="v1.26" state="alpha" >}}
+
+The Component SLIs feature allows enabling of a new metrics endpoint for each Kubernetes
+binary. This metric endpoint is exposed on the standard port of each binary at the endpoint
+`metrics/slis`. 
+
```
Comment (sftim) [inline content/en/docs/reference/instrumentation/slis.md:16]:
```suggestion
You must enable to the `ComponentSLIs` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
for every component from which you want to scrape SLI metrics.

```

**Review by sftim — COMMENTED** [review]
Thanks @logicalhan 

There's two kinds of feedback here.

- key things, which I hope you'll be willing to fix
- nits (marked as such). Fixing these is optional; for alpha features, we're happy to live with a few rough edges.

**Comment by sftim** [comment]
LGTM for SIG Instrumentation I think

There are a couple of things to fix for SIG Docs
/lgtm cancel

OK to re-LGTM once those are addressed.

File: content/en/docs/reference/instrumentation/slis.md (hunk 3 above)
Comment (logicalhan) [reply content/en/docs/reference/instrumentation/slis.md:55]: I actually do expect people who interact with prometheus metrics to be able to read the metric definitions. It's mostly declarative and easier to parse than most random code.

File: content/en/docs/reference/instrumentation/slis.md (hunk 4 above)
Comment (logicalhan) [reply content/en/docs/reference/instrumentation/slis.md:60]: There isn't syntax highlight for prometheus format, but I found shell colorizes very close to correctly.

File: content/en/docs/reference/instrumentation/slis.md (hunk 5 above)
Comment (logicalhan) [reply content/en/docs/reference/instrumentation/slis.md:81]: It colorizes pretty correctly under this syntax though. It looks nicer than if I omit here.

File: content/en/docs/reference/instrumentation/slis.md (hunk 3 above)
Comment (logicalhan) [reply content/en/docs/reference/instrumentation/slis.md:59]: removed

File: content/en/docs/reference/instrumentation/slis.md (hunk 5 above)
Comment (logicalhan) [reply content/en/docs/reference/instrumentation/slis.md]: removed

File: content/en/docs/reference/instrumentation/slis.md [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+weight: 20
+---
+
+<!-- overview -->
+
+{{< feature-state for_k8s_version="v1.26" state="alpha" >}}
+
+As an alpha feature, Kubernetes lets you configure Service Level Indicator (SLI) metrics 
+for each Kubernetes component binary. This metric endpoint is exposed on the serving 
+HTTPS port of each component, at the path `/metrics/slis`. You must enable to the
```
Comment (reylejano) [inline content/en/docs/reference/instrumentation/slis.md:16]: nit
```suggestion
HTTPS port of each component, at the path `/metrics/slis`. You must enable the 
```

**Comment by reylejano** [comment]
1.26 Doc PR for Kubernetes Component Health SLIs, enhancement issue https://github.com/kubernetes/enhancements/issues/3466

/label tide/merge-method-squash

This PR has an lgtm from a SIG Instrumentation tech lead, @dgrisonnet 

/approve

**Pull request merged**